packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/model.ts

```diff
@@ -156,19 +164,57 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
             },
           ],
         };
-      } else if (
+      }
+
+      const mappingsAreUnchanged =
+        // the source exists
+        Boolean(indices[source!]?.mappings?._meta?.migrationMappingPropertyHashes) &&
+        // ...and mappings are unchanged
+        !diffMappings(
+          /* actual */
+          indices[source!].mappings,
+          /* expected */
+          stateP.targetIndexMappings
+        );
+
+      if (mappingsAreUnchanged) {
+        const targetIndex = source!;
+
+        return {
+          ...stateP,
+          // We set a special flag that indicates we DO NOT need to reindex
+          controlState: 'OUTDATED_DOCUMENTS_SEARCH_OPEN_PIT',
+          sourceIndex: Option.none,
+          targetIndex,
+          sourceIndexMappings: indices[source!].mappings,
+          // Setting this to "undefined" will trigger UPDATE_TARGET_MAPPINGS
+          // which we always want to do for new versions
+          targetIndexCurrentMappings: undefined,
+          targetIndexMappings: mergeMigrationMappingPropertyHashes(
+            stateP.targetIndexMappings,
+            indices[source!].mappings
+          ),
+          // Point the current and version alias to the same index
+          versionIndexReadyActions: Option.some<AliasAction[]>([
+            { add: { index: source!, alias: stateP.currentAlias } },
+            { add: { index: source!, alias: stateP.versionAlias } },
+          ]),
```

Comment:
How/where do we remove the the old version alias? I'm guess this just repoints the existing .kibana alias and adds a new one for the current version.

Reply:
Assuming that the question is "at what stage in the migration to we update the version alias?",

We update the alias when in the [`MarkVersionIndexReady` state](https://github.com/elastic/kibana/blob/main/packages/core/saved-objects/core-saved-objects-migration-server-internal/src/next.ts#L188) and in the [`LegacyDeleteState` state](https://github.com/elastic/kibana/blob/main/packages/core/saved-objects/core-saved-objects-migration-server-internal/src/next.ts#L211).

The state -> actions mappings with their params are declared in the `next.ts` [file](https://github.com/elastic/kibana/blob/main/packages/core/saved-objects/core-saved-objects-migration-server-internal/src/next.ts).